### PR TITLE
WASM bin initial support

### PR DESCRIPTION
**Pull request by cgvwzq**
Still WIP but parses most WASM files I've tried (may require some code review).

TODO:
- add exports, tables and memories into symbols
- add support for custom sections

### CODE REVIEW

**Comment by Maijin** [comment]
Add bins in r2r https://github.com/radare/radare2-regressions/tree/master/bins + some tests https://github.com/radare/radare2-regressions/tree/master/t.formats like https://github.com/radare/radare2-regressions/blob/master/t.formats/nes/Pong

**Comment by cgvwzq** [comment]
Done! https://github.com/radare/radare2-regressions/pull/792

File: libr/bin/p/bin_wasm.c [hunk 1]
```diff
@@ -42,152 +57,203 @@ static RBinAddr *binsym(RBinFile *arch, int type) {
...
 
-static RList *sections(RBinFile *arch) {
-	RList *ret = NULL;
-	RBinSection *ptr = NULL;
-	// ut64 textsize, datasize, symssize, spszsize, pcszsize;
-	if (!arch->o->info) {
+static RList *symbols(RBinFile *arch) {
+
+	RBinWasmObj *bin;
+    RList *ret, *codes, *imports;
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: spaces vs tabs

File: libr/bin/p/bin_wasm.c [hunk 2]
```diff
@@ -42,152 +57,203 @@ static RBinAddr *binsym(RBinFile *arch, int type) {
...
+		ptr->ordinal = i;
+		i += 1;
+		r_list_append (ret, ptr);
+	}
+
+	// TODO: use custom section "name" if present
+	// TODO: exports, globals, tables and memories
+
+	return ret;
+
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: empty line

File: libr/bin/p/bin_wasm.c [hunk 3]
```diff
@@ -42,152 +57,203 @@ static RBinAddr *binsym(RBinFile *arch, int type) {
...
+}
+
+static RList *imports(RBinFile *arch) {
+
```
Comment (radare) [inline libr/bin/p/bin_wasm.c]: empty line

**Pull request merged**